pytradfri/device/__init__.py

```diff
@@ -134,50 +131,53 @@ class DeviceInfo:
         6: "AC (Mains) power",
         7: "Solar",
     }
-    ATTR_BATTERY = "9"
 
-    def __init__(self, device):
+    def __init__(self, device: Device) -> None:
         """Create object of class."""
         self._device = device
 
     @property
-    def manufacturer(self):
+    def manufacturer(self) -> str:
         """Human readable manufacturer name."""
-        return self.raw.get(DeviceInfo.ATTR_MANUFACTURER)
+        return self.raw["0"]
 
     @property
-    def model_number(self):
+    def model_number(self) -> str:
         """Return model identifier string (manufacturer specified string)."""
-        return self.raw.get(DeviceInfo.ATTR_MODEL_NUMBER)
+        return self.raw["1"]
 
     @property
-    def serial(self):
+    def serial(self) -> str:
         """Return serial string."""
-        return self.raw.get(DeviceInfo.ATTR_SERIAL)
+        return self.raw["2"]
 
     @property
-    def firmware_version(self):
+    def firmware_version(self) -> str:
         """Return current firmware version of device."""
-        return self.raw.get(DeviceInfo.ATTR_FIRMWARE_VERSION)
+        return self.raw["3"]
 
     @property
-    def power_source(self):
+    def power_source(self) -> int:
         """Power source."""
-        return self.raw.get(DeviceInfo.ATTR_POWER_SOURCE)
+        return self.raw["6"]
 
     @property
-    def power_source_str(self):
+    def power_source_str(self) -> Optional[str]:
         """Represent current power source."""
-        if DeviceInfo.ATTR_POWER_SOURCE not in self.raw:
+        if "6" not in self.raw:
             return None
+
         return DeviceInfo.VALUE_POWER_SOURCES.get(self.power_source, "Unknown")
 
     @property
-    def battery_level(self):
+    def battery_level(self) -> Optional[int]:
```

Comment:
Judging by the unit tests, this is optional today. Don't want to change that here.